apps/passport-server/src/services/generic-issuance/pipelines/LemonadePipeline.ts

```diff
@@ -864,15 +892,19 @@ export class LemonadePipeline implements BasePipeline {
     });
   }
 
-  private async getOrGenerateTicket(
-    ticketData: ITicketData
-  ): Promise<EdDSATicketPCD> {
+  private async getOrGenerateTicket<T extends EdDSATicketPCD | PODTicketPCD>(
+    ticketData: ITicketData,
+    ticketPCDType: T["type"]
+  ): Promise<T> {
     return traced(LOG_NAME, "getOrGenerateTicket", async (span) => {
       span?.setAttribute("ticket_id", ticketData.ticketId);
       span?.setAttribute("ticket_email", ticketData.attendeeEmail);
       span?.setAttribute("ticket_name", ticketData.attendeeName);
 
-      const cachedTicket = await this.getCachedTicket(ticketData);
+      const cachedTicket = await this.getCachedTicket<T>(
+        ticketData,
+        ticketPCDType
+      );
```

Comment:
this'll still be relevant at some point: https://linear.app/0xparc-pcd/issue/0XP-424/make-ticket-caching-logic-robust-to-new-ticket-pcd-version